File: client/src/main/java/net/treset/minecraftlauncher/sync/GetResponse.java
package net.treset.minecraftlauncher.sync;

import net.treset.mc_version_loader.json.GenericJsonParsable;
import net.treset.mc_version_loader.json.SerializationException;

import java.util.List;

public class GetResponse extends GenericJsonParsable {
    private int version;
    private List<String> difference;

    public GetResponse(int version, List<String> difference) {
        this.version = version;
        this.difference = difference;
    }

    public int getVersion() {
        return version;
    }

    public void setVersion(int version) {
        this.version = version;
    }

    public List<String> getDifference() {
        return difference;
    }

    public void setDifference(List<String> difference) {
        this.difference = difference;
    }

    public static GetResponse fromJson(String json) throws SerializationException {
        return GenericJsonParsable.fromJson(json, GetResponse.class);
    }
}

File: client/composeApp/src/commonMain/kotlin/net/treset/treelauncher/backend/sync/GetResponse.kt
package net.treset.treelauncher.backend.sync

import net.treset.mcdl.json.GenericJsonParsable
import net.treset.mcdl.json.SerializationException

class GetResponse(var version: Int, var difference: Array<String>) : GenericJsonParsable() {

    companion object {
        @Throws(SerializationException::class)
        fun fromJson(json: String?): GetResponse {
            return fromJson(json, GetResponse::class.java)
        }
    }
}
